PG-1997 - Add 17.7.1 release notes

Andriciuc opened the pull request.

Andriciuc committed:
Add 17.7.1 release notes

Andriciuc committed:
updated component versions

- added pg_tde component version
- removed adressed CVEs and SBOMs

Andriciuc committed:
update pgsm version to 2.3.1

Andriciuc committed:
add third party extensions intro paragraph

Andriciuc committed:
minor version update to pg_bouncer and PG commons

Andriciuc committed:
update upstream rn link and added addressed cves

Andriciuc committed:
Add telemetry deprecation as release highlight

ImTheKai commented on docs/release-notes/release-notes-v17.7.1.md:
```diff
@@ -0,0 +1,76 @@
+
+### Percona Telemetry Extension for PostgreSQL Deprecated
+
+The `percona_pg_telemetry` has been deprecated and replaced with a backwards compatibility stub.
```
The `percona_pg_telemetry` extension has... (reads better with extension added)

ImTheKai replied:
The `percona_pg_telemetry` extension has been deprecated and replaced with a backwards compatibility stub to avoid breaking existing installations.

Andriciuc replied:
Done! Thank you for the review!

Andriciuc committed:
Update Telemetry mention based on feedback

Andriciuc committed:
fix etcd mention in ha

- fix release note issue with component link
- fix release notes index macro issue
- update sboms to have pgversion tag
- updated tarballs to have pgversion tag

Andriciuc committed:
add 3.5 links as well and modify 3.x to 3.1.x to permit 3.5 increments

Andriciuc committed:
Add standalone package mention for pg_tde

Andriciuc committed:
update pg_tde standalone mention with better information

Andriciuc committed:
update based on dependency status change

Andriciuc committed:
Update release-notes-v17.7.1.md

Andriciuc committed:
Merge branch '17' into add-17.7.1-release-notes

Andriciuc committed:
Update main.yml

Andriciuc committed:
update version nr and dates
